core/drivers/crypto/versal/include/ipi.h

```diff
@@ -0,0 +1,96 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (C) Foundries Ltd. 2022 - All Rights Reserved
+ * Author: Jorge Ramirez <jorge@foundries.io>
+ */
+
+#ifndef IPI_H
+#define IPI_H
+
+#include <drivers/versal_mbox.h>
+
+struct versal_rsa_input_param {
+	uint64_t key_addr;
+	uint64_t data_addr;
+	uint32_t key_len;
+};
+
+struct versal_rsa_sign_param {
+	uint64_t sign_addr;
+	uint64_t hash_addr;
+	uint32_t hash_len;
+};
+
+struct versal_ecc_sign_param {
+	uint64_t hash_addr;
+	uint64_t priv_key_addr;
+	uint64_t epriv_key_addr;
+	uint32_t curve;
+	uint32_t hash_len;
+};
+
+struct versal_ecc_verify_param {
+	uint64_t hash_addr;
+	uint64_t pub_key_addr;
+	uint64_t signature_addr;
+	uint32_t curve;
+	uint32_t hash_len;
+};
+
+enum versal_aes_operation { ENCRYPT, DECRYPT };
```

Comment:
Avoid names that are too generic in the global scope. They may cause conflicts. `VERSAL_ENCRYPT`, `VERSAL_DECRYPT`.